tests/manage/pv_services/pv_encryption/test_rbd_pv_encryption_vaulttenantsa.py

```diff
@@ -0,0 +1,175 @@
+        log.info(
+            "Configuring kube auth method and csi-kms-connection-details configmap"
+        )
+        self.kms = vault_tenant_sa_setup_factory(
+            kv_version,
+            use_auth_path=use_auth_path,
+            use_vault_namespace=use_vault_namespace,
+        )
+        log.info("Test setup complete")
+
+    @tier1
+    def test_rbd_pv_encryption_vaulttenantsa(
+        self,
+        project_factory,
+        storageclass_factory,
+        multi_pvc_factory,
+        pod_factory,
+        kv_version,
+    ):
+        """
+        Test to verify creation and deletion of encrypted RBD PVC using vaulttenantsa method
+
+        """
+        # Create a project
+        proj_obj = project_factory()
+
+        # Create an encryption enabled storageclass for RBD
+        sc_obj = storageclass_factory(
+            interface=constants.CEPHBLOCKPOOL,
+            encrypted=True,
+            encryption_kms_id=self.kms.kmsid,
+        )
+
+        # Create serviceaccount in the tenant namespace
+        self.kms.create_tenant_sa(namespace=proj_obj.namespace)
+
+        # Create role in Vault
+        self.kms.create_vault_kube_auth_role(tenant_namespace=proj_obj.namespace)
+
+        # Create RBD PVCs with volume mode Block
+        pvc_size = 5
+        pvc_objs = multi_pvc_factory(
+            interface=constants.CEPHBLOCKPOOL,
+            project=proj_obj,
+            storageclass=sc_obj,
+            size=pvc_size,
+            access_modes=[
+                f"{constants.ACCESS_MODE_RWX}-Block",
+                f"{constants.ACCESS_MODE_RWO}-Block",
+            ],
+            status=constants.STATUS_BOUND,
+            num_of_pvc=3,
+            wait_each=False,
+        )
+
+        # Create pods
+        pod_objs = create_pods(
+            pvc_objs,
+            pod_factory,
+            constants.CEPHBLOCKPOOL,
+            pods_for_rwx=1,
+            status=constants.STATUS_RUNNING,
+        )
+
+        # Verify if the key is created in Vault
+        vol_handles = []
+        for pvc_obj in pvc_objs:
+            pv_obj = pvc_obj.backed_pv_obj
+            vol_handle = pv_obj.get().get("spec").get("csi").get("volumeHandle")
+            vol_handles.append(vol_handle)
+
+        # Check if encryption key is created in Vault
+        if kms.is_key_present_in_path(key=vol_handle, path=self.kms.vault_backend_path):
+            log.info(f"Vault: Found key for {pvc_obj.name}")
+        else:
+            raise ResourceNotFoundError(f"Vault: Key not found for {pvc_obj.name}")
+
+        # Verify whether encrypted device is present inside the pod and run IO
+        for vol_handle, pod_obj in zip(vol_handles, pod_objs):
+            if pod_obj.exec_sh_cmd_on_pod(
+                command=f"lsblk | grep {vol_handle} | grep crypt"
+            ):
+                log.info(f"Encrypted device found in {pod_obj.name}")
+            else:
+                log.error(f"Encrypted device not found in {pod_obj.name}")
```

Comment:
Don't we want to raise an exception here?

Reply:
Done